boards/stm32f3discovery/src/main.rs

```diff
@@ -607,6 +609,26 @@ pub unsafe fn reset_handler() {
     );
     stm32f303xc::adc::ADC1.set_client(adc);
 
+    // Kernel storage region, allocated with the storage_volume!
+    // macro in common/utils.rs
+    extern "C" {
+        /// Beginning on the ROM region containing app images.
+        static _sstorage: u8;
+        static _estorage: u8;
+    }
+
+    let nonvolatile_storage = components::nonvolatile_storage::NonvolatileStorageComponent::new(
+        board_kernel,
+        &stm32f303xc::flash::FLASH,
+        0x08038000,
+        0x8000,
```

Comment:
Given that these are somewhat arbitrary selections, it'd be nice to add a comment explaining how they were chosen (and that they're flexible).

Reply:
I agree. The problem I see with these arbitrary selections (this is how they are done in other boards too) is that nothing stops tock from loading apps over the userspace start address. Correct me if I am wrong. I think this could be solved in the kernel linker script by adding start and end symbols for this userspace region.

Reply:
Agreed. This part of the design got hung up a bit a little while ago in a discussion about application identifiers (i.e. how to map storage space to applications) I think. Agreed that a dedicated section would probably be better in the short term, but I think in the long term we'd have a richer concept of 'application flash storage'.